### PR TITLE
[libmagic] Add new port

### PR DESCRIPTION
**Pull request by NancyLi1013**
This library can be used to classify files according to magic number tests.
`libmagic `currently doesn't support Windows platform.

Related issue #7895 
The latest version **5.37** in [ftp://ftp.astron.com/pub/file/](url) cannot be reached, so I get the source from here https://github.com/threatstack/libmagic.

 No feature need to test.

### CODE REVIEW

File: ports/libmagic/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+vcpkg_install_make()
+
+vcpkg_copy_pdbs()
+
+if(VCPKG_LIBRARY_LINKAGE STREQUAL static)
+    file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/bin ${CURRENT_PACKAGES_DIR}/debug/bin)
+endif()
+
+# Handle copyright
+file(INSTALL ${SOURCE_PATH}/COPYING DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT}/copyright)
```
Comment (JackBoosY) [inline ports/libmagic/portfile.cmake]: should be
`file(INSTALL ${SOURCE_PATH}/COPYING DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)`

**Review by JackBoosY — CHANGES_REQUESTED** [review]

**Review by JackBoosY — APPROVED** [review]

**Comment by zrax** [comment]
The threatstack mirror is very old (5 years or so)...  If FTP can't be used, there is a much more up-to-date mirror on github at https://github.com/file/file

**Comment by NancyLi1013** [comment]
@zrax thanks for your info. I have updated.

**Review by dan-shaw — APPROVED** [review]

**Comment by dan-shaw** [comment]
LGTM

**Comment by ras0219-msft** [comment]
Please modify ci.baseline.txt to account for the windows failures

**Pull request merged**